### DISCUSSION TITLE
Error : undefined reference to '__device_dts_ord_18'

### DISCUSSION THREAD

**Discussion by sachinjairam** [zephyrproject-rtos/zephyr#42126] [Q&A] [answered]
Hai, I am trying to read ADC values for Nucleo-H743ZI board with Zephyr microROS.
But, while building the firmware an unknown error occurs.

![oie_xhJx2XTIHUXN](https://user-images.githubusercontent.com/44083276/150927234-bdc88a4e-cf27-454e-9bfc-74cace302f99.png)

I couldn't find any solution. So please advise me how to solve this problem.

Thank you,
Sachin

**Comment by FRASTM** [comment]
@sachinjairam could you please give more details about the application you are building
For example dev is the 
` const struct device *dev_adc = DEVICE_DT_GET(ADC_NODE);`
when building the ./samples/drivers/adc for the nucleo_h743zi target board which has an overlay made of 
```
/ {
	zephyr,user {
		/* adjust channel number according to pinmux in board.dts */
		io-channels = <&adc1 1>;
	};
};
```

**Reply by sachinjairam** [reply]
Hai, @FRASTM thanks for your reply.
I am trying to read ADC values from ADC1_INP15 of NUCLEO-H743ZI-Q board. here dev is the
```const struct device *dev_adc = DEVICE_DT_GET(ADC_NODE);```
and an overlay made of
```
/ {
	zephyr,user {
		/* adjust channel number according to pinmux in board.dts */
		io-channels = <&adc1 15>;
	};
};
```
Thank you.

**Comment by FRASTM** [comment]
Your talking about the _NUCLEO-H743ZI-Q_ board, do you have the same board definition (dts) as the NUCLEO-H743ZI including the nucleo_h743zi_q.overlay

**Reply by sachinjairam** [reply]
I am using NUCLEO-H743ZI-Q board with nucleo_h743zi.dts and nucleo_h743zi.overlay files.

**Reply by FRASTM** [reply]
I do not know what is the diff between the nucleo_h743zi and your NUCLEO-H743ZI-Q board
Based on the nucleo_h745zi_q definition that could be a matter of SMPS regulator

**Comment by FRASTM** [comment, marked as answer]
With that overlay
```
/ {
	zephyr,user {
		/* adjust channel number according to pinmux in board.dts */
		io-channels = <&adc1 15>;
	};
};
```
there is no pb to run the samples/drivers/adc :

```
*** Booting Zephyr OS build v3.0.0-rc1-92-g6393d650fc12*
ADC reading: 649                                        
ADC reading: 649                                        
ADC reading: 648                                        
ADC reading: 648                                        
ADC reading: 645                                        

```
Also runs the tests/drivers/adc/adc_api using the ADC1 channel 15 for the nucleo_h743zi board
